tests/unit/s2n_key_material_test.c

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "utils/s2n_key_material.h"
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+#include "tls/s2n_cipher_suites.h"
+#include "tls/s2n_connection.h"
+#include "tls/s2n_crypto.h"
+
+S2N_RESULT helper_generate_test_data(struct s2n_blob *test_data)
+{
+    struct s2n_stuffer test_data_stuffer = { 0 };
+    EXPECT_SUCCESS(s2n_stuffer_init(&test_data_stuffer, test_data));
+    for (int i = 0; i < S2N_MAX_KEY_BLOCK_LEN; i++) {
+        s2n_stuffer_write_uint8(&test_data_stuffer, i);
+    }
+    return S2N_RESULT_OK;
+}
+
+S2N_RESULT helper_validate_key_material(struct s2n_key_material *key_material, struct s2n_blob *test_data_blob,
+        uint8_t mac_size, uint8_t key_size, uint8_t iv_size)
+{
+    /* confirm that the data is copied to key_material */
+    EXPECT_EQUAL(memcmp(test_data_blob->data, key_material->data, test_data_blob->size), 0);
```

Comment:
Generally a good idea to compare sizes before comparing memory, unless it's not possible for the sizes not to agree